ssg/templates.py

```diff
@@ -285,25 +247,51 @@ def get_lang_for_rule(self, rule_id, rule_title, template, language):
         # definition ID so that the build system matches the generated
         # check with the rule.
         template_vars["_rule_id"] = rule_id
-        # checks and remediations are processed with a custom YAML dict
-        local_env_yaml = self.env_yaml.copy()
-        local_env_yaml["rule_id"] = rule_id
-        local_env_yaml["rule_title"] = rule_title
-        local_env_yaml["products"] = self.env_yaml["product"]
+        # Checks and remediations are processed with a custom YAML dict
+        local_env_yaml = {"rule_id": rule_id, "rule_title": rule_title}
+        if extra_env is not None:
+            local_env_yaml.update(extra_env)
```

Comment:
This will lead to dragging rule_id and rule_title even further down the pipe. Ideally I would like to organize this context in a more opaque and organized way, but it'll come later.

Reply:
I think that I agree with both of you, so I would do it the other way around - Evgeny is correct that the `render_lang_file` doesn't know anything specific about rules, and it just has that dictionary with everything that it can use without going into details.
And Jan is correct as well - so why not to move the addition of the product from the "env yaml" one level above, so modifications of this dictionary is localized in one method, and the rendering method becomes more abstract?

Reply:
It can't be done ATM because of https://github.com/ComplianceAsCode/content/blob/9c9f7c2ad405139bd5bc009dfaf3bb443e21b148/ssg/build_sce.py#L169